gcovr/tests/test_gcov_parser.py

```diff
@@ -276,3 +280,39 @@ def test_pathologic_codeline(capsys):
         'IndexError',
         '(ERROR) Exiting',
         'run gcovr with --gcov-ignore-parse-errors')
+
+
+def check_and_raise(number, total, exc_raised, queue_full):
+    queue_full.wait()
+    if number == 0:
+        raise Exception("Number == 0")
+    exc_raised.wait()
+    total[0] += 1
+
+
+@pytest.mark.parametrize('threads', [1, 2, 4, 8])
+def test_pathologic_threads(capsys, threads):
+    mutable = [0]
+    queue_full = Event()
+    exc_raised = Event()
+    with pytest.raises(Exception) as excinfo:
+        with Workers(threads, lambda: {'total': mutable, 'exc_raised': exc_raised, 'queue_full': queue_full}) as pool:
+            for extra in range(0, 10000):
+                pool.add(check_and_raise, extra)
+
+            # Queue is filled
+            queue_full.set()
+
+            # Wait until the exception has been completed
+            while not pool.exceptions:
+                time.sleep(0)
```

Comment:
Is `sleep(0)` guaranteed to yield from this thread and let other threads run? Or might this be a `while True: pass` style infinite loop?

Reply:
It is (https://docs.python.org/3/library/sched.html) but it was a pain to find! Waiting like this is obviously not great, but I didn't want to complicate the workers code to make the test cleaner.

Reply:
Good find! You can leave this as it is, or add a comment.

Reply:
Added a comment.